### PR TITLE
Extract ratings logic to ratings vm and add tracks

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This extracts rating logic to its own vm and adds a track event.

## Testing Instructions
Make sure everything still works from the testing steps in https://github.com/Automattic/pocket-casts-android/pull/868 with a change that ratings load when the podcast is opened instead of when podcast details are expanded to make it consistent with iOS.

Test Tracks
1. With the rating feature flag enabled, open a podcast
2. Tap on the rating stars view
3. Notice that you see the event: `Tracked: rating_stars_tapped, Properties: {"uuid":"...` where uuid is the uuid of the selected podcast.

## Merge Instructions

Since the feature is guarded with a feature flag, it is safe to merge it into `main`:

- Review this PR
- Change base branch to main 
- Merge the PR.

### CODE REVIEW

**Review by mchowning — APPROVED** [review]
This is working well. Nice work! 👍

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt [hunk 1]
```diff
@@ -142,14 +144,13 @@ class PodcastFragment : BaseFragment(), Toolbar.OnMenuItemClickListener, Corouti
         podcastUuid: String,
         expanded: Boolean,
         userInitiated: Boolean,
-    ) -> Unit = { podcastUuid, expanded, userInitiated ->
+    ) -> Unit = { _, expanded, userInitiated ->
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt]: Removed in dbba614

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastRatingsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        ) : RatingState()
+
+        object Error : RatingState()
+    }
+
+    sealed class Star {
+        object FilledStar : Star()
+        object HalfStar : Star()
+        object BorderedStar : Star()
+    }
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastRatingsViewModel.kt]: 🤦‍♀️ changed in [733c4a7](https://github.com/Automattic/pocket-casts-android/pull/880/commits/733c4a753ecffd6d604496cf56f62b1b5ee6872d)

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/StarRatingView.kt [hunk 3]
```diff
@@ -21,50 +24,74 @@ import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
...
+}
+
+@Composable
+private fun Content(
+    state: RatingState.Loaded,
+    onClick: () -> Unit
+) {
     Row(
         modifier = Modifier
             .fillMaxWidth()
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/StarRatingView.kt]: Removed in [847980e](https://github.com/Automattic/pocket-casts-android/pull/880/commits/847980eeda0d910f7949da35deffcfbb6578d715)

**Pull request merged**